java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```
@@ -194,6 +182,48 @@ public void release(String lockId) {
+                            .setName(secretName.toString())
+                            // Label format is https://cloud.google.com/compute/docs/labeling-resources#requirements
+                            .putLabels(label, labelValue)
+                            .build())
+                    .setUpdateMask(FieldMask.newBuilder()
+                            .addPaths("labels")
+                            .build())
+                    .build());
+        } catch (Exception e) {
+            log.severe(String.format("Cannot put the label of the version on the secret %s due exception: %s", secretName.toString(), e));
+        }
+    }
+
+    private static void disableOldSecretVersions(SecretManagerServiceClient client, SecretName secretName, SecretVersion exceptVersion) {
+        try {
+            SecretManagerServiceClient.ListSecretVersionsPage enabledVersions = client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+                            .setFilter("state:ENABLED")
+                            .setParent(secretName.toString())
+                            // Reduce the page, as each version will be disabled one by one
+                            .setPageSize(NUMBER_OF_VERSIONS_TO_RETRIEVE)
+                            .build())
+                    .getPage();
+
+            // Disable secret version, just the first page
+            enabledVersions.getValues().forEach(i -> {
+                if (!i.getName().equals(exceptVersion.getName())) {
+                    client.disableSecretVersion(DisableSecretVersionRequest.newBuilder()
+                            .setName(i.getName())
+                            .build());
+
+                    log.info(String.format("Property: %s, disabled version %s", secretName, i.getName()));
+                }
+            });
+        } catch (Exception e) {
+            log.severe(String.format("Cannot disable old versions from secret %s due exception: %s", secretName.toString(), e));
```

i think leave suppressing the exception to the caller